### ISSUE TITLE
Made an applet; question

### ISSUE THREAD

**Issue by junelva** [MaxVerevkin/wl-gammarelay-rs#16]
Hi, thanks for rewriting this tool in Rust. The original was causing me trouble before and I noticed its Arch package was out-of-date. In searching for a replacement, I found your tool. I like it, so I made a GUI applet to configure it. https://github.com/junelva/wl-gammarelay-applet

I'm new to DBus and was having difficulty researching this online. In my applet, after rapidly scrolling the sliders for about five seconds, the screen ceases to be updated. I was wondering if you knew any reason that might be. I tried limiting the rate at which I call updates, but the issue remains. Thanks.

**Comment by MaxVerevkin** [comment]
Cool project!

I got this panic while testing it:

```
thread 'main' panicked at src/main.rs:207:66:
rust: get session: InputOutput(Os { code: 24, kind: Uncategorized, message: "Too many open files" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I think the problem here is that you create a new dbus connection each time a value changes, and you run out of file descriptors.

So what you should do is keep a single `zbus::Connection` and accumulate the pending changes between dbus calls.

**Comment by junelva** [comment]
Thanks for that tip. It worked! Cheers.

**Comment by junelva** [comment]
Thank you! I've pushed it a bit further, and want to thank you for creating a reliable wlroots color adjustment server in Rust. With the applet I made, my desktop ergonomics went way up. I can't explain why I need to frequently adjust screen inversion, temperature, brightness, and gamma, but calibrating a display for comfort by sight is my preferred mechanism, and your server enables me to do this effortlessly. Thanks again. :)

By the way, I looked at your other projects. I think I'll try river. Your `wayrs` libraries look very appealing! I may attempt to create similar applets using that instead of Slint, since their licensing stance is geared toward industrial competition rather than freedom.
